exec/java-exec/src/test/java/org/apache/drill/test/RestClientFixture.java

```diff
@@ -0,0 +1,115 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.drill.test;
+
+import com.google.common.base.Preconditions;
+import io.restassured.RestAssured;
+import io.restassured.http.ContentType;
+import org.apache.drill.exec.server.rest.StatusResources;
+
+import javax.annotation.Nullable;
+
+import static io.restassured.RestAssured.when;
+
+/**
+ * Represents a client for the Drill Rest API.
+ */
+public class RestClientFixture {
+  public static final String REST_API_PORT = "drill.exec.http.port";
+
+  /**
+   * A builder for the rest client.
+   */
+  public static class Builder {
+    private ClusterFixture cluster;
+
+    public Builder(ClusterFixture cluster) {
+      this.cluster = Preconditions.checkNotNull(cluster);
+    }
+
+    public RestClientFixture build() {
+      return new RestClientFixture(cluster);
+    }
+  }
+
+
+  private RestClientFixture(ClusterFixture cluster) {
+    int port = cluster.config.getInt(REST_API_PORT);
+    String address = cluster.drillbits().iterator().next().getContext().getEndpoint().getAddress();
+
+    RestAssured.port = port;
+    RestAssured.basePath = "/";
+    RestAssured.baseURI = "http://" + address;
```

Comment:
Are these global variables? How will this work in Surefire, when multiple tests run concurrently, if someone uses this mechanism to, say, test HTTPS support in one test while this runs in another test?

Reply:
Yeah the rest-assured library uses global variables, which I don't like either. But it gives some nice additionally functionality for testing. The global configuration shouldn't impact us though because we run tests concurrently by forking separate jvms, and running tests sequentially within each jvm. If you see this being a larger issue later when we potentially run tests concurrently within a jvm, I can flip back to using the jersey-client which does not have this problem.

Reply:
I reverted to using the jersey client so global variables aren't used anymore.